Sources/App/Core/PackageContributors.swift

```diff
@@ -0,0 +1,132 @@
+    var authors : [Author]
+    var numberOfContributors : Int
+}
+
+enum PackageContributors {
+
+    /// Extracts the possible authors of the package according to the number of commits.
+    /// A contributor is considered an author when the number of commits is at least a 60 percent
+    /// of the maximum commits done by a contributor.
+    /// - Parameters:
+    ///   - gitCacheDirectoryPath: path to the cache directory where the clone of the package is stored
+    ///   - packageID: the UUID of the package
+    /// - Returns: PackageAuthors
+    static func extract(gitCacheDirectoryPath: String, packageID: UUID?) throws -> PackageAuthors {
+        let contributorsHistory = try GitHistoryLoader.loadContributorsHistory(gitCacheDirectoryPath: gitCacheDirectoryPath, packageID: packageID)
+        let authors = primaryContributors(candidates: contributorsHistory, threshold: 0.6)
+
+        return PackageAuthors(authors: authors.map { Author(name: $0.name) },
+                              numberOfContributors: contributorsHistory.count - authors.count)
+    }
+
+
+    struct Contributor {
+        /// Total number of commits
+        let numberOfCommits: Int
+        let name: String
+    }
+
+    /// Loads the contributors history from a Git repository
+    struct GitHistoryLoader {
+
+        static func loadContributorsHistory(gitCacheDirectoryPath: String, packageID: UUID?) throws -> [Contributor] {
+            do {
+                let commitHistory = try queryGitHistory(gitCacheDirectoryPath: gitCacheDirectoryPath, packageID: packageID)
+                return try parseGitHistory(log: commitHistory)
+            } catch {
+                throw AppError.analysisError(packageID, "loadContributorsHistory failed: \(error.localizedDescription)")
+            }
+        }
+
+        /// Gets the git history in a string log
+        private static func queryGitHistory(gitCacheDirectoryPath: String, packageID: UUID?) throws -> String {
+
+            if !Current.fileManager.fileExists(atPath: gitCacheDirectoryPath) {
+                throw AppError.cacheDirectoryDoesNotExist(packageID, gitCacheDirectoryPath)
+            }
+
+            // attempt to shortlog
+            do {
+                return try Current.git.shortlog(gitCacheDirectoryPath)
+            } catch {
+                throw AppError.shellCommandFailed("gitShortlog",
+                                                  gitCacheDirectoryPath,
+                                                  "queryGitHistory failed: \(error.localizedDescription)")
+            }
+        }
+
+        /// Parses the result of queryGitHistory into a collection of contributors
+        private static func parseGitHistory(log: String) throws -> [Contributor] {
+            var committers = [Contributor]()
+
+            for line in log.components(separatedBy: .newlines) {
+                let log = line.split(whereSeparator: { $0 == " " || $0 == "\t"})
+
+                if (log.count > 2) {
```

Comment:
nit: this would probably read nicer as a `guard`.

```
guard log.count > 2 else {
  continue
}

// continue your logic for handling identifier/number extraction.
```